security-analysis-api/src/main/java/com/powsybl/security/SecurityAnalysis2.java

```diff
@@ -0,0 +1,292 @@
+            Objects.requireNonNull(interceptors, "Interceptor list should not be null");
+            return provider.run(network, workingStateId, detector, filter, computationManager, parameters, contingenciesProvider, interceptors);
+        }
+
+        public CompletableFuture<SecurityAnalysisResult> runAsync(Network network, LimitViolationFilter filter,
+                                                                  ComputationManager computationManager) {
+            return runAsync(network, network.getVariantManager().getWorkingVariantId(), new DefaultLimitViolationDetector(), filter, computationManager, SecurityAnalysisParameters.load(), new EmptyContingencyListProvider(), Collections.emptyList());
+        }
+
+        public CompletableFuture<SecurityAnalysisResult> runAsync(Network network, ComputationManager computationManager) {
+            return runAsync(network, LimitViolationFilter.load(), computationManager);
+        }
+
+        public CompletableFuture<SecurityAnalysisResult> runAsync(Network network) {
+            return runAsync(network, new LocalComputationManagerFactory().create());
+        }
+
+        public CompletableFuture<SecurityAnalysisResultWithLog> runAsyncWithLog(Network network,
+                                                                                String workingStateId,
+                                                                                LimitViolationDetector detector,
+                                                                                LimitViolationFilter filter,
+                                                                                ComputationManager computationManager,
+                                                                                SecurityAnalysisParameters parameters,
+                                                                                ContingenciesProvider contingenciesProvider,
+                                                                                List<SecurityAnalysisInterceptor> interceptors) {
+            Objects.requireNonNull(network, "Network should not be null");
+            Objects.requireNonNull(workingStateId, "WorkingVariantId should not be null");
+            Objects.requireNonNull(detector, "LimitViolation detector should not be null");
+            Objects.requireNonNull(filter, "LimitViolation filter should not be null");
+            Objects.requireNonNull(computationManager, "ComputationManager should not be null");
+            Objects.requireNonNull(contingenciesProvider, "Contingencies provider should not be null");
+            Objects.requireNonNull(parameters, "Sensitivity analysis parameters should not be null");
+            Objects.requireNonNull(interceptors, "Interceptor list should not be null");
+            return provider.runWithLog(network, workingStateId, detector, filter, computationManager, parameters, contingenciesProvider, interceptors);
+        }
```

Comment:
I think you should remove all the `WithLog` method: as proposed last time, we would probably return an object that contain the result of the simulation and a report. To be discussed with the others

Reply:
Isn't it already the purpose of SecurityAnalysisResultWithLog?

Reply:
Maybe we should just remove SecurityAnalysisResultWithLog and move logBytes to SecurityAnalysisResult (and up to an impl to use it or not) so that the API will be cleaner?

Reply:
There was one question with this approach : how do you handle serialization of this ? Do you want to serialize it in the JSON representing the result or not ?

Reply:
Issue #1602 was created to keep these comments